File: isValid.js
function isValid(s) {
        let arr = []
        for(let c = 0; c < s.length; c++){
            if(s[c]=="}"&&arr[arr.length-1]=="{")
                arr.pop()
            else if(s[c]=="]"&&arr[arr.length-1]=="[")
                arr.pop()
            else if(s[c]==")"&&arr[arr.length-1]=="(")
                arr.pop()
            else if(s[c]=="]"||s[c]=="}"||s[c]==")")
                return false;
            else    
                arr.push(s[c])
        }
        return arr.length == 0;
    }

File: is_valid.py
def is_valid(string: str) -> bool:
    pairs = {"{":"}", "(":")", "[":"]"}
    stack = []
    for ch in string:
        if ch in pairs.keys():
            stack.append(ch)
        elif len(stack) == 0:
            return False
        elif ch != pairs.get(stack.pop()):
            return False
    return len(stack) == 0

print(is_valid("{()[]"))
